src/activities/content/ContentHtmlFileEntity.js

```diff
@@ -0,0 +1,133 @@
+	 * @returns {string|null} Description html of the content-htmlFile item
+	 */
+	descriptionRichText() {
+		const descriptionSubEntity = ContentHelperFunctions.getDescriptionSubEntity(this._entity);
+		if (!descriptionSubEntity) {
+			return null;
+		}
+		return descriptionSubEntity.properties.html || '';
+	}
+
+	/**
+	 * @returns {string|null} Description text of the content-htmlFile item
+	 */
+	descriptionText() {
+		const descriptionSubEntity = ContentHelperFunctions.getDescriptionSubEntity(this._entity);
+		if (!descriptionSubEntity) {
+			return null;
+		}
+		return descriptionSubEntity.properties.text || '';
+	}
+
+	/**
+	 * @returns {string} Title of the content-htmlFile item
+	 */
+	title() {
+		return this._entity && this._entity.properties && this._entity.properties.title;
+	}
+
+	/**
+	 * Updates the html file to have the given description
+	 * @param {string} richText rich text description to set on the html file
+	 */
+	async setHtmlFileDescription(richText) {
+		if (!this._entity) {
+			return;
+		}
+		const action = this._entity.getActionByName(Actions.content.updateDescription);
+		if (!action) {
+			return;
+		}
+
+		const fields = [{ name: 'description', value: richText }];
+		await performSirenAction(this._token, action, fields);
+	}
+
+	/**
+	 * Updates the html file to have the given title
+	 * @param {string} title Title to set on the html file
+	 */
+	async setHtmlFileTitle(title) {
+		if (!this._entity) {
+			return;
+		}
+		const action = this._entity.getActionByName(Actions.content.updateTitle);
+		if (!action) {
+			return;
+		}
+
+		const fields = [{ name: 'title', value: title }];
+		await performSirenAction(this._token, action, fields);
+	}
+
+	/**
+	 * Updates the html file content with the given html
+	 * @param {html} html to set on the html file
+	 */
+	async setHtmlFileHtml(html) {
+		if (!this._entity) {
+			return;
+		}
+		const action = this._entity.getActionByName(Actions.htmlFile.updateHtmlContent);
+		if (!action) {
+			return;
+		}
+
+		const fields = [{ name: 'htmlContent', value: html }];
+		await performSirenAction(this._token, action, fields);
+	}
+
+	/**
+	 * Deletes the html file
+	 */
+	async deleteHtmlFile() {
+		if (!this._entity) {
+			return;
+		}
+		const action = this._entity.getActionByName(Actions.htmlFile.deleteHtmlFile);
+		if (!action) {
+			return;
+		}
+
+		await performSirenAction(this._token, action);
+		this.dispose();
+	}
+
+	/**
+	 * Checks if content html file properties passed in match what is currently stored
+	 * @param {object} contentHtmlFile Object containing html file specific properties
+	 */
+	equals(contentHtmlFile) {
+		const diffs = [
+			[this.title(), contentHtmlFile.title],
+		];
+		for (const [left, right] of diffs) {
+			if (left !== right) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	* @returns {string} Returns URL of the files home API
+	*/
+	getFilesHref() {
+		if (!this._entity || !this._entity.hasLinkByRel(Rels.Files.file)) {
+			return;

```

Comment:
Something similar here. IMO we should be more explicit about what we're returning (maybe return null) instead of an empty return that will get set to `undefined`, but this also doesn't match the expected return type based on the comment above the function

Reply:
I agree with you. I'm going to make a separate PR which will address this and test stuff.

Reply:
Wondering if this should be singular (`getFileHref`) as opposed to plural. The href points to a specific file entry, correct?